src/test/java/com/hillel/items_exchange/security/SecurityConfigIntegrationTest.java

```diff
@@ -89,34 +101,83 @@ public void loginWithNotValidUserIsOk() throws Exception {
     }
 
     @Test
+    @Transactional
     public void createAdvertisementWithoutTokenIsUnauthorized() throws Exception {
         mockMvc.perform(post("/adv"))
                 .andExpect(status().isUnauthorized());
     }
 
     @Test
+    @Transactional
     @DataSet("database_init.yml")
     public void createAdvertisementWithValidTokenWithoutAdvertisementDtoIsBadRequest() throws Exception {
-        final String token = obtainToken(validLoginDto);
+        final String validToken = "Bearer " + obtainToken(validLoginDto);
+
         mockMvc.perform(post("/adv")
-                .header("Authorization", "Bearer " + token))
+                .header("Authorization", validToken))
                 .andExpect(status().isBadRequest());
     }
 
-//    TODO: Create new dataset for current test and fix the test
-//    @Test
-//    @Transactional
-//    @DataSet("database_init.yml")
-//    @ExpectedDataSet(value = "advertisement/create.yml", ignoreCols = {"created", "updated"})
-//    public void createAdvertisementWithValidTokenAndValidAdvertisementDtoIsOk() throws Exception {
-//        final String token = obtainToken(validLoginDto);
-//        mockMvc.perform(post("/adv")
-//                .header("Authorization", "Bearer " + token)
-//                .content(asJsonString(nonExistDto))
-//                .contentType(MediaType.APPLICATION_JSON)
-//                .accept(MediaType.APPLICATION_JSON))
-//                .andExpect(status().isCreated());
-//    }
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    @ExpectedDataSet(value = "security/create-advertisement.yml", ignoreCols = {"created", "updated"})
+    public void createAdvertisementWithValidTokenAndValidAdvertisementDtoIsOk() throws Exception {
+        final String validToken = "Bearer " + obtainToken(validLoginDto);
+
+        mockMvc.perform(post("/adv")
+                .header("Authorization", validToken)
+                .content(asJsonString(nonExistDto))
+                .contentType(MediaType.APPLICATION_JSON)
```

Comment:
Would you mind updating all test cases here with such constants?

Reply:
Performed